vsphere/datadog_checks/vsphere/types.py

```diff
@@ -0,0 +1,62 @@
+from typing import Dict, List, Optional, Pattern, Tuple, Type, TypedDict
+
+# CONFIG ALIASES
+from pyVmomi import vim
+
+ResourceFilterConfig = TypedDict('ResourceFilterConfig', {'resource': str, 'property': str, 'patterns': List[str]})
+MetricFilterConfig = Dict[str, List[str]]
+
+InstanceConfig = TypedDict(
+    'InstanceConfig',
+    {
+        'host': str,
+        'username': str,
+        'password': str,
+        'ssl_verify': bool,
+        'ssl_capath': str,
+        'tls_ignore_warning': bool,
+        'collection_level': int,
+        'collection_type': str,
+        'use_guest_hostname': bool,
+        'max_historical_metrics': int,
+        'threads_count': int,
+        'metrics_per_query': int,
+        'batch_property_collector_size': int,
+        'collect_events': bool,
+        'collect_tags': bool,
+        'tags_prefix': str,
+        'excluded_host_tags': List[str],
+        'tags': List[str],
+        'refresh_infrastructure_cache_interval': int,
+        'refresh_metrics_metadata_cache_interval': int,
+        'refresh_tags_cache_interval': int,
+        'resource_filters': List[ResourceFilterConfig],
+        'metric_filters': MetricFilterConfig,
+        'collect_per_instance_filters': MetricFilterConfig,
+    },
+)
+
+# CHECK ALIASES
+MetricName = str
+CounterId = int
+
+InfrastructureDataItem = TypedDict(
+    'InfrastructureDataItem',
+    {
+        'name': str,
+        'runtime.host': Type[vim.ManagedEntity],
```

Comment:
Looks like this entry should refer to an instance instead of a class, as seen by using [here](https://github.com/DataDog/integrations-core/pull/6036/files#diff-a28471e1e1032abae47738867b3c6870R76), no?

```suggestion
        'runtime.host': vim.ManagedEntity,
```

Reply:
Good catch.